Extending use cases (#23)
temporalio/documentation-legacy: docs/use-cases-periodic.md

---
id: use-cases-periodic

> title: Periodic Execution

Reviewer: This section seems very similar to monitoring right?

Reviewer: I think you might be able to merge this in the other section. Not sure.

Author: Yes, actually I excluded it from the navigation pane and left the file to keep the existing links working. Maybe we should place permanent redirects instead.

Reviewer: That works for me.

Reviewer: Can you handle the redirect or should I have Ruslan help?

Author: I tried the [client redirects plugin](https://v2.docusaurus.io/docs/using-plugins/#docusaurusplugin-client-redirects):

1. It wants me to delete the old page (okay)
2. It doesn't seem to play nicely with versioning: at least, I have to specify `/next` in the URLs, or otherwise it doesn't recognize the URL because `/docs/use-cases-monitoring` is not in the current version.
3. `yarn start` doesn't seem to respect the redirect, while `yarn build` does.

Should I go forward with this or does Ruslan have experience with redirects in docusaurus? Or maybe merge as-is and add redirects when the version is current?

Reviewer: Ok so we've spent a lot of time (surprisingly) discussing this. Our decision is to not maintain backwards compatibility and blow away the old versions. That means you don't need to worry about redirecting, just update to the state you want for the currents docs.

:)

sidebar_label: Periodic Execution
---

## Motivation

Periodic execution of business logic, also known as a **distributed cron** engine.

> Typical examples of periodic service orchestration include:

Reviewer: I think this change is actually not accurate as some of the listed cases are periodic execution but not all of them are periodic orchestration.

- Monthly report generation.
- Weekly newsletter delivery.
- Daily loyalty point accumulation per customer.
- Once-per-minute health checks of a production deployment.

## Benefits

> Temporal provides guaranteed execution, sophisticated error handling, flexible retry policies, and visibility into execution history for periodic workflow execution.

Reviewer: ```suggestion
Temporal provides guaranteed execution, sophisticated error handling, flexible retry policies, and visibility into execution history for periodic workflow executions.
```

> Scalability is another crucial advantage. Many use cases require periodic execution for a large number of entities. At Uber, some applications run recurring workflows for each customer. Imagine 100+ million parallel cron jobs that don't require a separate batch processing framework.

Reviewer: ```suggestion
Scalability is another crucial advantage of using Temporal for periodic execution. Many use cases require periodic execution for a large number of entities. At Uber, some applications run recurring workflows for each customer. Imagine 100+ million parallel cron jobs that don't require a separate batch processing framework.
```

## Example

> You can specify a cron expression when starting a workflow with a client. The snippet below schedules hourly executions of a sample workflow:

Reviewer: ```suggestion
You can specify a [cron expression](https://en.wikipedia.org/wiki/Cron#CRON_expression) when starting a workflow. The snippet below schedules hourly executions of a sample workflow:
```

```go
workflowOptions := client.StartWorkflowOptions{

>     ID:           workflowID,

Reviewer: ```suggestion
    ID:           "someWorkflowID",
```
